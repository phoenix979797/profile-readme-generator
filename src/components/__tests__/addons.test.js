import React from "react"
import toJson from "enzyme-to-json"
import { shallow } from "enzyme"

import Addons from "../addons"

describe("Addons", () => {
  const dataInput = {
    title: "",
    subtitle: "A passionate frontend developer from India",
    currentWork: "",
    currentLearn: "",
    collaborateOn: "",
    helpWith: "",
    ama: "",
    contact: "",
    funFact: "",
    twitterBadge: false,
    visitorsBadge: false,
    badgeStyle: "flat",
    badgeColor: "0e75b6",
    badgeLabel: "Profile views",
    githubProfileTrophy: false,
    githubStats: false,
    githubStatsOptions: {
      theme: "",
      titleColor: "",
      textColor: "",
      bgColor: "",
      hideBorder: false,
      cacheSeconds: null,
      locale: "en",
    },
    topLanguages: false,
    topLanguagesOptions: {
      theme: "",
      titleColor: "",
      textColor: "",
      bgColor: "",
      hideBorder: false,
      cacheSeconds: null,
      locale: "en",
    },
    devDynamicBlogs: false,
    mediumDynamicBlogs: false,
    rssDynamicBlogs: false,
  };
  const socialInput = {
    github: "",
    dev: "",
    linkedin: "",
    codepen: "",
    stackoverflow: "",
    kaggle: "",
    codesandbox: "",
    fb: "",
    instagram: "",
    twitter: "",
    dribbble: "",
    behance: "",
    medium: "",
    youtube: "",
    codechef: "",
    hackerrank: "",
    codeforces: "",
    leetcode: "",
    topcoder: "",
    hackerearth: "",
    geeks_for_geeks: "",
    discord: "",
    rssurl: "",
  };

  let mockHandleCheckChange;
  let mockHandleDataChange;
  beforeEach(() => {
    mockHandleCheckChange = jest.fn();
    mockHandleDataChange = jest.fn();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("renders correctly", () => {
    const addOnComponent = shallow(<Addons
      data={dataInput}
      social={socialInput}
      handleCheckChange={mockHandleCheckChange}
      handleDataChange={mockHandleDataChange}
    />);

    expect(toJson(addOnComponent)).toMatchSnapshot();
  });
});