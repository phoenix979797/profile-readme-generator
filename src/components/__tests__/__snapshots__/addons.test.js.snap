// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Addons renders correctly 1`] = `
<div
  className="flex justify-center items-start flex-col w-full px-2 sm:px-6 mb-10"
>
  <div
    className="text-xl sm:text-2xl font-bold font-title mt-2 mb-2"
  >
    Add-ons
  </div>
  <AddonsItem
    Options={
      <CustomizeOptions
        CustomizationOptions={
          <CustomizeBadge
            badgeOptions={
              Object {
                "badgeColor": "0e75b6",
                "badgeLabel": "Profile views",
                "badgeStyle": "flat",
              }
            }
            githubName=""
            onBadgeUpdate={[Function]}
          />
        }
        title="Customize Badge"
      />
    }
    inputChecked={false}
    inputId="visitors-count"
    onInputChange={[Function]}
  >
    display visitors count badge
  </AddonsItem>
  <AddonsItem
    inputChecked={false}
    inputId="github-profile-trophy"
    onInputChange={[Function]}
  >
    display github trophy
  </AddonsItem>
  <AddonsItem
    Options={
      <CustomizeOptions
        CustomizationOptions={
          <CustomizeGithubStatsBase
            onUpdate={[Function]}
            options={
              Object {
                "bgColor": "",
                "cacheSeconds": null,
                "hideBorder": false,
                "locale": "en",
                "textColor": "",
                "theme": "",
                "titleColor": "",
              }
            }
            prefix="stats"
          />
        }
        title="Customize Github Stats Card"
      />
    }
    inputChecked={false}
    inputId="github-stats"
    onInputChange={[Function]}
  >
    display github profile stats card
  </AddonsItem>
  <AddonsItem
    Options={
      <CustomizeOptions
        CustomizationOptions={
          <CustomizeGithubStatsBase
            onUpdate={[Function]}
            options={
              Object {
                "bgColor": "",
                "cacheSeconds": null,
                "hideBorder": false,
                "locale": "en",
                "textColor": "",
                "theme": "",
                "titleColor": "",
              }
            }
            prefix="top-lang"
          />
        }
        title="Customize Top Skills Card"
      />
    }
    inputChecked={false}
    inputId="top-languages"
    onInputChange={[Function]}
  >
    display top skills
  </AddonsItem>
  <AddonsItem
    inputChecked={false}
    inputId="twitter-badge"
    onInputChange={[Function]}
  >
    display twitter badge
  </AddonsItem>
  <AddonsItem
    inputChecked={false}
    inputId="dev-dynamic-blogs"
    onInputChange={[Function]}
  >
    display latest dev.to blogs dynamically (GitHub Action)
  </AddonsItem>
  <AddonsItem
    inputChecked={false}
    inputId="medium-dynamic-blogs"
    onInputChange={[Function]}
  >
    display latest medium blogs dynamically (GitHub Action)
  </AddonsItem>
  <AddonsItem
    inputChecked={false}
    inputId="rss-dynamic-blogs"
    onInputChange={[Function]}
  >
    display latest blogs from your personal blog dynamically (GitHub Action)
  </AddonsItem>
</div>
`;
